src/api.py

```diff
@@ -0,0 +1,62 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+import requests
+import tempfile 
+
+class Response:
+    """
+    this object stores all the information regarding the URL which our API
+    has hit 
+    1) Location of file
+    2) Content type of the file
+    3) Size of the file
+    4) URL of the file
+    """
+    def __init__(self, filename, content_type, size, url):
+        self.filename = filename
+        self.content_type = content_type
+        self.size = size
+        self.url = url
+
+
+def get_request(url):
+    return requests.get(url)
+
+
+def Filewriter(filename,content):
+    with open(filename,'wb') as f:
+        f.write(content)
+
+def fetch(url):
+    """
+    Takes URL as parameter and write down the content inside it
+    to a temporary location and return a `Response` kind of object
+    """
+    r = get_request(url=url)
+
+    temp = tempfile.NamedTemporaryFile(delete=False)
+    filename = temp.name
+
+    Filewriter(filename=filename , content=r.content)
+
+    resp = Response(filename=filename,
+    content_type=r.headers['content-type'],
+    size=int(r.headers['content-length']),
+    url=url)
```

Comment:
when having args on multiple lines , put one kwarg per line and indent accordingly to promote readability.